python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -243,71 +243,45 @@ def get_partitions_def_for_dimension(self, dimension_name: str) -> PartitionsDef
                 return dim_def.partitions_def
         check.failed(f"Invalid dimension name {dimension_name}")
 
-    def get_partition(
+    # We override the default implementation of `has_partition_key` for performance.
+    def has_partition_key(
         self,
-        partition_key: str,
+        partition_key: Union[MultiPartitionKey, str],
         current_time: Optional[datetime] = None,
         dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
-    ) -> Partition:
-        if not isinstance(partition_key, MultiPartitionKey):
-            partition_key = self.get_partition_key_from_str(partition_key)
-        partition_key = cast(MultiPartitionKey, partition_key)
-
-        if partition_key.keys_by_dimension.keys() != set(self.partition_dimension_names):
-            raise DagsterUnknownPartitionError(
-                f"Invalid partition key {partition_key}. The dimensions of the partition key are"
-                " not the dimensions of the partitions definition."
-            )
-
-        partitions_by_dimension: Dict[str, Partition] = {}
+    ) -> bool:
+        partition_key = (
+            partition_key
+            if isinstance(partition_key, MultiPartitionKey)
+            else self.get_partition_key_from_str(partition_key)
+        )
```

Comment:
I think we still need the removed check:
```
if partition_key.keys_by_dimension.keys() != set(self.partition_dimension_names):
            return False
```
in case of the partition key's dimension names don't match

Reply:
ah right